### PR TITLE
Devise branch PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this branch, me and @rica213;

- Removed current_user method in ApplicationController.

- Installed and setup devise, and added below functionalities;
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.

- Fixed linter errors

Looking forward to hearing from you.

Thanks.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (prolajumokeoni) [inline app/models/user.rb]: - [x] Kindly make sure that `:confirmable` is enabled in the user model, Currently, when I signup I am redirected to the  homepage without my email being confirmed please make sure a user confirms email before they can access the home page 👍🏿

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -11,4 +11,6 @@
 
     <%= form.submit 'Create Post', class: 'post-btn' %>
```
Comment (prolajumokeoni) [inline app/views/posts/new.html.erb:12]: - [x] Kindly make sure a user can create post successfully and see the post created, when I click the create post I can't see new posts created 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Team
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Application allows for registering a new user. ✅ 

 -  `Current_user` method in `ApplicationController` is removed 👏🏿 
  



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/new.html.erb (hunk 2 above)
Comment (Whoistolu) [reply app/views/posts/new.html.erb:12]: - Well resolved 👍🏽 
Screenshot 2023-05-31 at 16 44 41

File: app/models/user.rb (hunk 1 above)
Comment (Whoistolu) [reply app/models/user.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi Team,

Great job implementing the recommendation of the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mertkantaroglu** [comment]
Thank you @Whoistolu 👍

**Pull request merged**